### PR TITLE
Migration to null safety

### PR DESCRIPTION
**Pull request by afermin**
Reopened:

Hi there,

I just migrated to null safety the plugin regarding this policygenius#10

I hope you can take a look and left some comments if needed. @maxnovak @longhorn499

Best.

## Motivation and Context

It's required to be able to migrate to dart null safety.

[10](https://github.com/policygenius/optimizely-plugin/issues/10)

## Type of Change:

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Maintenance or Refactor

## Checklist

- [ ] I added tests to cover my changes
- [ ] I rebased changes with master so that they can be merged easily
- [ ] I have updated the documentation accordingly (if applicable)

### CODE REVIEW

**Review by maxnovak — COMMENTED** [review]
Could you please undo the format changes, which are unrelated to your PR?
e.g. the double to single quotes & tab changes

**Comment by afermin** [comment]
> Could you please undo the format changes, which are unrelated to your PR?
> e.g. the double to single quotes & tab changes

Done

**Comment by afermin** [comment]
Looks like the flutter version should updated in the CI

File: example/lib/main.dart [hunk 1]
```diff
@@ -28,7 +28,7 @@ class _MyAppState extends State<MyApp> {
     // Uses the Optimizely example:
     // https://docs.developers.optimizely.com/full-stack/docs/example-datafile
     final dataFile =
-        await DefaultAssetBundle.of(context).loadString('assets/datafile.json');
+    await DefaultAssetBundle.of(context).loadString('assets/datafile.json');
```
Comment (longhorn499) [inline example/lib/main.dart:31]: Revert spacing

File: example/lib/main.dart [hunk 2]
```diff
@@ -46,9 +46,9 @@ class _MyAppState extends State<MyApp> {
     String priceFilterFlag;
     // Platform messages may fail, so we use a try/catch PlatformException.
     var platform =
-        Theme.of(context).platform.toString().split('.')[1].toLowerCase();
+    Theme.of(context).platform.toString().split('.')[1].toLowerCase();
```
Comment (longhorn499) [inline example/lib/main.dart:49]: +1

File: example/lib/main.dart [hunk 3]
```diff
@@ -72,14 +72,14 @@ class _MyAppState extends State<MyApp> {
     String minPriceVariable;
     Map<String, dynamic> variables;
     var platform =
-        Theme.of(context).platform.toString().split('.')[1].toLowerCase();
+    Theme.of(context).platform.toString().split('.')[1].toLowerCase();
```
Comment (longhorn499) [inline example/lib/main.dart:75]: +1

Comment (afermin) [reply example/lib/main.dart]: Done

File: example/lib/main.dart (hunk 2 above)
Comment (afermin) [reply example/lib/main.dart]: Done

File: example/lib/main.dart (hunk 1 above)
Comment (afermin) [reply example/lib/main.dart]: Done

**Comment by longhorn499** [comment]
> Looks like the flutter version should updated in the CI

Yep, I think updating it to `2.0.4` (or any newer version past the latest major release) should do it there. You can update that in `.github/workflows/cd.yaml` and if that passes locally I can merge this in and do the release. Not sure if updating that on this branch will cause the action here to run with the higher version 🤔 but I can pull it locally if not and merge / follow-up and fix the action

**Comment by afermin** [comment]
> > Looks like the flutter version should updated in the CI
> 
> Yep, I think updating it to `2.0.4` (or any newer version past the latest major release) should do it there. You can update that in `.github/workflows/ci.yaml` and if that passes locally I can merge this in and do the release. Not sure if updating that on this branch will cause the action here to run with the higher version (or it will just run w/ the action settings on `main`) 🤔 but I can pull it locally if not and merge / follow-up and fix the action

Thanks for that. Already changed it and it worked

**Review by longhorn499 — APPROVED** [review]
lgtm! 🌠 Thanks for contributing

**Comment by afermin** [comment]
> lgtm! 🌠 Thanks for contributing

My pleasure. Thank you for letting me do that!

**Comment by longhorn499** [comment]
> > lgtm! 🌠 Thanks for contributing
> 
> My pleasure. Thank you for letting me do that!

Of course, thanks again. It has been published. There are notes in the release on GitHub, but I will follow-up with an update in the `CHANGELOG` as well as fix some warnings I found when running `dart pub publish --dry-run`

**Pull request merged**